tests/test_curate_preferences.py

```
@@ -297,5 +438,213 @@ def test_invalid_utf8_fails_closed(self):
+class PreferencePurityNineteenRegression(unittest.TestCase):
+    """Bind the nineteen known impure pairs to deterministic decisions.
+
+    The committed corpus under tests/fixtures/preference-purity/ mirrors the
+    real raw ffpc corpus line-for-line in layout and decision (10 files, 42
+    records, 19 impure: 7 repaired + 12 excluded, 23 pure controls), as
+    measured by a read-only scan of outputs/raw/2026-08-17/
+    failure-as-fuel-preference-cascade. Raw stays immutable and unreferenced.
+    """
+
+    @classmethod
+    def setUpClass(cls):
```

Ruff (0.16.1)

[warning] Missing return type annotation for classmethod `setUpClass`

Add return type annotation: `None`

(ANN206)